File: src/Wordki.Core/Domain/Word.cs
using System;

namespace Wordki.Core
{
    public class Word
    {
        public long Id { get; set; }
        public long GroupId { get; set; }
        public Group Group { get; set; }
        public long UserId { get; set; }
        public string Language1 { get; set; }
        public string Language2 { get; set; }
        public string Language1Example { get; set; }
        public string Language2Example { get; set; }
        public byte Drawer { get; set; }
        public bool IsVisible { get; set; }
        public int State { get; set; }
        public bool IsSelected { get; set; }
        public ushort RepeatingCounter { get; set; }
        public DateTime LastRepeating { get; set; }
        public string Comment { get; set; }
        public DateTime LastChange { get; set; }

        public Word()
        {
            Id = 0;
            Language1 = string.Empty;
            Language2 = string.Empty;
            Language1Example = string.Empty;
            Language2Example = string.Empty;
            IsVisible = true;
            State = int.MaxValue;
            Comment = string.Empty;
        }
    }
}

File: client/src/app/common/models/Word.ts
export class Word {
  id: string = '';
  groupId: string = '';
  language1: string = '';
  language1Example: string = '';
  language2: string = '';
  language2Example: string = '';
  drawer: number = 0;
  isVisible: boolean = true;
  repeatingCounter: number = 0;

  increaseDrawer() {
    this.drawer++;
    if (this.drawer > 4) {
      this.drawer = 4;
    }
  }

  resetDrawer(): void {
    this.drawer = 0;
  }

  deserialize(input: any): this {
    Object.assign(this, input);
    return this;
  }

  clone(): Word {
    let word = new Word();
    word.id = this.id;
    word.groupId = this.groupId;
    word.drawer = this.drawer;
    word.language1 = this.language1;
    word.language1Example = this.language1Example;
    word.language2 = this.language2;
    word.language2Example = this.language2Example;
    word.isVisible = this.isVisible;
    word.repeatingCounter = this.repeatingCounter;
    return word;
  }
}